pandas/tests/indexes/test_interval.py

```diff
@@ -491,6 +649,207 @@ def testcontains(self):
         assert not i.contains(20)
         assert not i.contains(-20)
 
+        # Why are there two tests called test contains? this should be cleared up a little, no?
+
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_interval_covers_interval_updated_behavior(self):
+
+        # class Interval:
+        #     def covers(self, other: Interval) -> bool
+
+        assert     Interval(1, 3).covers(Interval(1.5, 2.5))
+        assert     Interval(1, 3).covers(Interval(1, 2))
+        assert     Interval(1, 3).covers(Interval(2, 3))
+        assert not Interval(1, 3).covers(Interval(0.5, 2.5))
+        assert not Interval(1, 3).covers(Interval(1.5, 3.5))
+
+        assert     Interval(1, 3, closed='right').covers(Interval(1, 3, closed='right'))
+        assert not Interval(1, 3, closed='right').covers(Interval(1, 3, closed='left'))
+        assert not Interval(1, 3, closed='right').covers(Interval(1, 3, closed='both'))
+
+        assert not Interval(1, 3, closed='left').covers(Interval(1, 3, closed='right'))
+        assert     Interval(1, 3, closed='left').covers(Interval(1, 3, closed='left'))
+        assert not Interval(1, 3, closed='left').covers(Interval(1, 3, closed='both'))
+
+        assert     Interval(1, 3, closed='both').covers(Interval(1, 3, closed='right'))
+        assert     Interval(1, 3, closed='both').covers(Interval(1, 3, closed='left'))
+        assert     Interval(1, 3, closed='both').covers(Interval(1, 3, closed='both'))
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_interval_covers_intervalIndex_updated_behavior(self):
+
+        # class Interval:
+        #     def covers(self, other: IntervalIndex) -> IntegerArray1D
```

Comment:
Ah, OK, will first try to read the discussion there before commenting further, as I at least disagree with the tests here (do you have the idea it already reached some consensus?) 

Still, I think it will be beneficial to keep the PRs and discussion in the PR manageable to leave it out here but put in separate one.